### PR TITLE
Moar tests

### LINKED ISSUE #223: Increase unit test coverage in native-metrics
`lib/pre-build.js` and `lib/upload.js` have less unit test coverage than the rest of this repo.

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Refactored code to make it easier to test
 * Add full suite of unit tests

## Links
Closes #223

### CODE REVIEW

File: tests/unit/pre-build.tap.js [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+      const expectedData = Buffer.from('testing', 'utf-8')
+      nock('https://download.newrelic.com/').get(/.*/).reply(200, expectedData)
+
+      preBuild.download('test', (err) => {
+        t.match(err.message, /Failed to unzip.*/)
+        t.end()
+      })
+    })
+
+    /* TODO: not sure how to test when res is a stream
```
Comment (jmartin4563) [inline tests/unit/pre-build.tap.js:214]: Do we still want to save this for later, or remove?

Comment (bizob2828) [reply tests/unit/pre-build.tap.js:214]: i wanted to keep i can provide more context or i can remove, thoughts?

Comment (jmartin4563) [reply tests/unit/pre-build.tap.js:214]: I'm cool with keeping if we plan on actually enabling the test at some point, otherwise I'd say remove

Comment (bizob2828) [reply tests/unit/pre-build.tap.js]: i just removed it

**Review by jmartin4563 — APPROVED** [review]
you're gonna have to merge main in to get the required checks, but this is good to go!

**Pull request merged**